### ISSUE TITLE
Regression: Process.StartInfo throws InvalidOperationException

### ISSUE THREAD

**Issue by heaths** [dotnet/corefx#37674]
Caused by PR #1167, calling `Process.StartInfo` on a process you did not start throws an `InvalidOperationException`. This is a regression from .NET Framework, which probably had the underlying issue #1100.

We rely on this information to copy the environment variables from another process and have never had a problem with inconsistency. The problems of getting other processes' `StartInfo` are and have been documented in the official API documentation; therefore, changing the behavior in Core seems unnecessary.

**Comment by danmosemsft** [comment]
The docs say
```
If you did not use the Start method to start a process, the StartInfo property does not reflect the parameters used to start the process. For example, if you use GetProcesses to get an array of processes running on the computer, the StartInfo property of each Process does not contain the original file name or arguments used to start the process.
```

The original bug report (like these docs) clearly states that the environment variables are wrong. You're reporting that for you, they're right. We need to resolve this question before deciding whether to change code again.

**Comment by heaths** [comment]
It says "does not reflect the parameters used to start the process", but not necessarily true of environment variables. It also doesn't talk about whether elevation is needed, which for some information it is (we run elevated as an installer of per-machine packages).

**Comment by danmosemsft** [comment]
Right but what motivated the original report #1100? Was that just speculation or based on actual report? It would be nice to know when or how often they are wrong before we just open up the API documented to "Usually work" 😺

**Comment by heaths** [comment]
In principle, I agree. But it is still a regression. I understand that .NET Core is able to make certain changes given its SxS nature, but even if `StartInfo` was inconsistent in the .NET Framework, porting code to .NET Core will at least be consistently inconsistent.

@davkean's [comment](https://github.com/dotnet/corefx/issues/1100#issuecomment-78758990) has merit, however. Given that, I was able to workaround this in our code. We were getting `StartInfo` unconditionally and rarely used it. In the code path where we did use it, it was for a `Process` we did indeed start. So I suppose we can close this one unless you think more conversation is warranted.
